lib/autoload/kuroko2/workflow/engine.rb

```diff
@@ -119,6 +120,7 @@ def process_next(node, token)
           token.mark_as_finished
           unless token.parent
             token.job_instance.touch(:finished_at)
+            Notifier.notify(:back_to_normal, token.job_instance) if token.context['RETRYING']
             Notifier.notify(:finished, token.job_instance)
```

Comment:
When retried, should "finished" notification be sent?

Reply:
It is necessary for when the job has enabled `Notify all event to Slack/Hipchat/Webhook` and disabled `Notify when back to normal` option.

Reply:
Sending both "back to normal" and "finish" notification with checking"  notification is duplicated when the job is enabled both "Notify all event to Slack/Hipchat/Webhook" and " Notify when back to normal", isn't it?

Reply:
Sorry, I understand. I fixed it in https://github.com/cookpad/kuroko2/pull/93/commits/370977be056f5f50ece9982bd53f2de3eeebaf11.